lib/Runtime/Debug/DebugContext.h

```diff
@@ -51,11 +51,22 @@ namespace Js
         void SetHostDebugContext(HostDebugContext * hostDebugContext);
 
         void SetDebuggerMode(DebuggerMode mode);
+
         bool IsDebugContextInNonDebugMode() const { return this->debuggerMode == DebuggerMode::NotDebugging; }
         bool IsDebugContextInDebugMode() const { return this->debuggerMode == DebuggerMode::Debugging; }
         bool IsDebugContextInSourceRundownMode() const { return this->debuggerMode == DebuggerMode::SourceRundown; }
         bool IsDebugContextInSourceRundownOrDebugMode() const { return IsDebugContextInSourceRundownMode() || IsDebugContextInDebugMode(); }
 
+        bool IsDebuggerRecording() const { return this->IsDebugContextInDebugMode() && this->isDebuggerRecording; }
+        void SetIsDebuggerRecording(bool isDebuggerRecording)
+        {
+            if (this->IsDebugContextInDebugMode())
```

Comment:
See other comment.

---
In reply to: [130155689](https://github.com/Microsoft/ChakraCore/pull/3447#discussion_r130155689) [](ancestors = 130155689)